rust-connectors/sinks/sql/src/db.rs

```diff
@@ -0,0 +1,482 @@
+
+    fn bind_value<'a, 'b>(
+        query: Query<'a, DB, <DB as HasArguments<'a>>::Arguments>,
+        value: &'b Value,
+    ) -> anyhow::Result<Query<'a, DB, <DB as HasArguments<'a>>::Arguments>>;
+}
+
+#[async_trait]
+impl Insert<Postgres> for Db {
+    async fn insert<'c, E: Executor<'c, Database = Postgres>>(
+        conn: E,
+        table: String,
+        values: Vec<Value>,
+    ) -> anyhow::Result<()> {
+        let columns = values.iter().map(|v| v.column.as_str()).join(",");
+        let values_clause = (1..=values.len()).map(|i| format!("${}", i)).join(",");
+
+        let sql = format!(
+            "INSERT INTO {} ({}) VALUES ({})",
+            table, columns, values_clause
+        );
+        debug!("sending sql: {}", &sql);
+        let mut query = sqlx::query(&sql);
+        for value in values {
+            query = match Self::bind_value(query, &value) {
+                Ok(q) => q,
+                Err(err) => {
+                    error!("Unable to bind {:?}. Reason: {:?}", value, err);
+                    return Err(err);
+                }
+            }
+        }
+        query.execute(conn).await?;
+        Ok(())
+    }
+
+    fn bind_value<'a, 'b>(
+        query: Query<'a, Postgres, PgArguments>,
+        value: &'b Value,
+    ) -> anyhow::Result<Query<'a, Postgres, PgArguments>> {
+        let query = match value.type_ {
+            Type::Bool => query.bind(bool::from_str(&value.raw_value)?),
+            Type::Char => query.bind(i8::from_str(&value.raw_value)?),
+            Type::SmallInt => query.bind(i16::from_str(&value.raw_value)?),
+            Type::Int => query.bind(i32::from_str(&value.raw_value)?),
+            Type::BigInt => query.bind(i64::from_str(&value.raw_value)?),
+            Type::Float => query.bind(f32::from_str(&value.raw_value)?),
+            Type::DoublePrecision => query.bind(f64::from_str(&value.raw_value)?),
+            Type::Text => query.bind(value.raw_value.clone()),
+            Type::Bytes => query.bind(value.raw_value.as_bytes().to_vec()),
+            Type::Numeric => query.bind(Decimal::from_str(&value.raw_value)?),
+            Type::Timestamp => query.bind(chrono::NaiveDateTime::parse_from_str(
+                &value.raw_value,
+                NAIVE_DATE_TIME_FORMAT,
+            )?),
+            Type::Date => query.bind(chrono::NaiveDate::from_str(&value.raw_value)?),
+            Type::Time => query.bind(chrono::NaiveTime::from_str(&value.raw_value)?),
+            Type::Uuid => query.bind(uuid::Uuid::from_str(&value.raw_value)?),
+            Type::Json => query.bind(serde_json::Value::from_str(&value.raw_value)?),
+        };
+        Ok(query)
+    }
+}
+
+#[async_trait]
+impl Insert<Sqlite> for Db {
+    async fn insert<'c, E: Executor<'c, Database = Sqlite>>(
+        conn: E,
+        table: String,
+        values: Vec<Value>,
+    ) -> anyhow::Result<()> {
+        let columns = values.iter().map(|v| v.column.as_str()).join(",");
```

Comment:
encapsulate logic such that it's possible to unit test without creating database connection

Reply:
Done